plugins/embedding.cc

```diff
@@ -286,6 +286,37 @@ struct EmbeddingDataset::Itl
         return result;
     }
 
+    struct EmbeddingRowStream : public RowStream {
+
+        EmbeddingRowStream(const EmbeddingDataset::Itl* source) : index(0), source(source)
+        {
+         
+        }
+
+        virtual std::shared_ptr<RowStream> clone() const{
```

Comment:
Why do we need to clone the row streams?  Let's at least make sure that we understand what the requirements are in a comment, eg where should the cloned stream point to?  The start?  The current element?